### PR TITLE
docs: add edit mode for stories in docs page

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532714909/pulses/5898424178

API:
- parameters.docs.liveEdit structure:
```typescript
liveEdit?: {
  scope?: Record<string, unknown>;
  isEnabled?: boolean;
};
```

`scope` can be any component or helper function.
so, it can be `someText` var that is a string
and it can be a component
whatever which is a key-value obj that you want the react-live's preview to know

The **only** thing needs to be done to enable LiveEdit for an existing component is simply removing the Canvas import in the MDX
If you'd like to disable for specific story (e.g. the Overview story), just set in its CSF the `isEnabled: false`

Mar-31-2024 20-29-08

### CODE REVIEW

File: packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { Canvas, useOf } from "@storybook/blocks";
+import { ComponentProps, FC, useMemo, useState } from "react";
+import styles from "./CanvasWrapper.module.scss";
+
+type CanvasWrapper = ComponentProps<typeof Canvas>;
+
+const CanvasWrapper: FC<CanvasWrapper> = ({ of }) => {
+  const [open, setOpen] = useState<boolean>(false);
+
+  const { story } = useOf(of || "story", ["story"]);
```
Comment (talkor) [inline packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx:10]: Can you explain pls?

File: packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  const toggleCodeAction = useMemo(
+    () => ({
+      title: "Toggle code editor",
```
Comment (talkor) [inline packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx:14]: Suggestion: call it "Play"

File: packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      onClick: () => setOpen(prev => !prev)
+    }),
+    []
+  );
+
+  const liveEditEnabledForStory = story.parameters.docs?.liveEdit?.enableLiveEdit;
```
Comment (talkor) [inline packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx:20]: Suggestion: is*

File: packages/core/src/storybook/decorators/withLiveEdit/withLiveEdit.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    setDirty(true);
+  };
+
+  useEffect(() => {
+    setCode(code => {
+      try {
+        return formatCode(code);
+      } catch (e) {
+        console.error(e);
+        return originalCode.current;
```
Comment (talkor) [inline packages/core/src/storybook/decorators/withLiveEdit/withLiveEdit.tsx:41]: Nice fallback

File: packages/core/src/storybook/decorators/withLiveEdit/withLiveEdit.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        <Story />
+      )}
+      {shouldAllowLiveEdit &&
+        createPortal(
+          <LiveEditor
+            placeholder={"Insert code here"}
+            code={code}
+            theme={vscodeDark}
+            extensions={[langs.tsx()]}
+            style={{ fontSize: 13 }}
```
Comment (talkor) [inline packages/core/src/storybook/decorators/withLiveEdit/withLiveEdit.tsx:67]: On purpose?

**Review by talkor — APPROVED** [review]
This is freaking amazing!!!! 👑 💯 💪🏼

File: packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx (hunk 2 above)
Comment (YossiSaadi) [reply packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx:14]: Just like that? Play?
It opens and shut the live editor, not sure about Play

File: packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx (hunk 3 above)
Comment (YossiSaadi) [reply packages/core/src/storybook/components/canvas-wrapper/CanvasWrapper.tsx:20]: liveEdit.isEnabled?

**Pull request merged**